src/components/EntitlementsPanelForm/Fields/Fields.tsx

```diff
@@ -0,0 +1,118 @@
+import { Select } from "@canonical/react-components";
+import { useQueryClient } from "@tanstack/react-query";
+import { useFormikContext } from "formik";
+
+import type { EntityEntitlement } from "api/api.schemas";
+import { useGetResources } from "api/resources/resources";
+import CleanFormikField from "components/CleanFormikField";
+import FormikSubmitButton from "components/FormikSubmitButton";
+import { Endpoint } from "types/api";
+
+import { Label } from "../types";
+
+type Props = {
+  entitlements: EntityEntitlement[];
+};
+
+const Fields = ({ entitlements }: Props) => {
+  const queryClient = useQueryClient();
+  const { values, setFieldValue, setFieldTouched } =
+    useFormikContext<EntityEntitlement>();
+  const { data: getResourcesData, isFetching: isGetResourcesFetching } =
+    useGetResources(
+      { entityType: values.entity_type },
+      { query: { enabled: !!values.entity_type } },
+    );
+  const resources = getResourcesData?.data.data || [];
+
+  return (
+    <div className="panel-table-form__fields">
+      <CleanFormikField
+        component={Select}
+        label={Label.ENTITY}
+        name="entity_type"
+        options={[
+          {
+            disabled: true,
+            label: "Select a resource type",
+            value: "",
+          },
+        ].concat(
+          [
+            ...new Set(
```

Comment:
Are the options in this component using a Set because there can be duplicates?

Reply:
It just occurred to me that this is because these are generated form the list of entitlements, not something specific to the entity_types.